trlx/models/modeling_ppo.py

```diff
@@ -410,10 +410,18 @@ def __init__(
         super().__init__(base_model.config)
 
         # The branch is defined by the last `num_layers_unfrozen` layers of the pretrained model
-        decoder_blocks = deepcopy(hf_get_decoder_blocks(base_model))
-        self.decoder_blocks = nn.ModuleList(list(decoder_blocks)[-num_layers_unfrozen:])
-        self.final_norm = deepcopy(hf_get_decoder_final_norm(base_model))
-        self.lm_head = deepcopy(hf_get_lm_head(base_model))
+
+        decoder_blocks = hf_get_decoder_blocks(base_model)[-num_layers_unfrozen:]
+        final_norm = hf_get_decoder_final_norm(base_model)
+        lm_head = hf_get_lm_head(base_model)
+
+        with deepspeed.zero.GatheredParameters(
```

Comment:
Should this be dependent on whether we are using zero3 (zero3_init)?

Reply:
@maxreciprocate

Reply:
@Dahoas Not necessary, this context would be a noop when zero3 is not enabled and would still work correctly, see comparisons runs https://wandb.ai/sorry/trlx-references/reports/fix-ppo-zero3-v-main--Vmlldzo0NjY3MDYy